device/common/include/traccc/clusterization/device/impl/aggregate_cluster.ipp

```diff
@@ -94,25 +95,24 @@ inline void aggregate_cluster(
         for (char i = 0; i < 2; ++i) {
             var[i] /= totalWeight;
         }
-        const auto pitch = this_module.pixel.get_pitch();
-        var = var + point2{pitch[0] * pitch[0] / static_cast<scalar>(12.),
-                           pitch[1] * pitch[1] / static_cast<scalar>(12.)};
+        var = var + point2{pitch_x * pitch_x / static_cast<scalar>(12.),
+                           pitch_y * pitch_y / static_cast<scalar>(12.)};
     }
 
     /*
      * Fill output vector with calculated cluster properties
      */
     out.local = mean;
     out.variance = var;
-    out.surface_link = this_module.surface_link;
+    out.surface_link = det_descr.surface_link().at(module_link);
     out.module_link = module_link;
     // Set a unique identifier for the measurement.
     out.measurement_id = link;
     // Adjust the output object for 1D surfaces.
-    if (this_module.pixel.dimension == 1) {
+    if (det_descr.dimensions().at(module_link) == 1) {
         out.meas_dim = 1;
-        out.local[1] = 0.f;
-        out.variance[1] = this_module.pixel.variance_y;
+        out.local[1] = scalar{0.5} * pitch_y;
+        out.variance[1] = det_descr.variance_y().at(module_link);
```

Comment:
Isn't this always going to be equal to `pitch_y * pitch_y / 12.f`? I'm curious if we actually need store these variances separately from the pitches?

Reply:
It's a fair question... Currently this variance comes from the follow type of declaration in the ODD digitization JSON file:

```
                    "segmentation": {
                        "binningdata": [
                            {
                                "bins": 384,
                                "max": 48,
                                "min": -48,
                                "option": "open",
                                "type": "equidistant",
                                "value": "binX"
                            },
                            {
                                "bins": 1,
                                "max": 54,
                                "min": -54,
                                "option": "open",
                                "type": "equidistant",
                                "value": "binY"
                            }
                        ]
                    },
                    "thickness": 0.25,
                    "threshold": 0.01,
                    "charge-smearing": {
                        "type": "Gauss",
                        "mean": 0.0,
                        "stddev": 0.015
                    },
                    "variances": [
                        {
                            "index": 0,
                            "rms": [
                                0.0037812280934304,
                                0.0001779164740582928
                            ]
                        },
                        {
                            "index": 1,
                            "rms": [
                                972.7376098632812
                            ]
                        }
                    ]
```

Where the I/O code currently takes that RMS value as-is for the variance. But I could be reasonably convinced that we could do it in a different / simpler way. :thinking:

Reply:
Indeed, the RMS of $972$ is equal to $\frac{54 - (-54)}{\sqrt{12}}$ in that example, although the JSON file has a few more significant digits... @asalzburger does it ever make sense for us to store the variance explicitly, i.e. is there ever a case in which it is not trivially computable from the pitch?

Reply:
Coming back to this. Is what you say actually true though? :confused:

The variance is calculated as `pitch^2/sqrt(12)` at the moment.

https://github.com/acts-project/traccc/blob/main/core/include/traccc/clusterization/impl/measurement_creation.ipp#L103-L104

In this case, the "pitch" would be `54 - ( -54)`. I.e. `108`. But the variance is then `108^2 / sqrt(12)`. But that is not 972. :confused:

Then again, the "variance" for this, per se, should be `108`. Not sure why it should be any bigger than that. :confused:

I'm happy to go with whatever math you guys prefer. The current PR just transcribes the logic that we currently use (since #581). I was about to open a PR that would change that logic separately from this PR, but now I'm not sure anymore what we really want here. :thinking: Would it be okay to not worry about this variance at all? Since the track finding should anyway respect that it's a 1D measurement, no? Or does the variance still play a role for 1D measurements as well? (I really don't know...)

Reply:
I think the vairance of 972 is 108*108/12 as the variance is L^2/12 ? :thinking:  

> Or does the variance still play a role for 1D measurements as well?

Yes we need it for both pixel and strip.

The hard-coded variances are only for digitization and we may not want to use them in the `measurement_creation`.

Instead, the measurement covariance should be determined by the cluster shape. This is rather easy for 1D strip where the variance is L^2/12. But pixel? we may need some R&D to do this properly...

Reply:
As far as I know, ATHENA does this with super fancy NN but I really want to have a less complicated method

Reply:
Ah.. what the second index here does is assuming that the particle is intersecting the center of strip and taking the rms based on the entire length of strip. That's why the rms is exceptionally high for the second index. 

So basically we can pretend the strip does the 2-dim measurement :thinking: but I am not sure if it is OK to use the second index measurement in the actual tracking as the rms is too high

Reply:
The `variance_y` is completely removed from the code by now.